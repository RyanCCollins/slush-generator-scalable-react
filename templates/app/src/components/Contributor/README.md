## Contributor Component
A component that ...

### Example

```js
<Contributor />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
