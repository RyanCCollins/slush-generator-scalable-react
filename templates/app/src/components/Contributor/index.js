import React, { PropTypes } from 'react';
import { Avatar } from 'components';
import cssModules from 'react-css-modules';
import Heading from 'grommet-udacity/components/Heading';
import Box from 'grommet-udacity/components/Box';
import Paragraph from 'grommet-udacity/components/Paragraph';
import Anchor from 'grommet-udacity/components/Anchor';
import SocialGithubIcon from 'grommet-udacity/components/icons/base/SocialGithub';
import styles from './index.module.scss';

const Contributor = ({
  person,
}) => (
  <Box
    className={styles.contributor}
    align="center"
    justify="center"
    size="large"
  >
    <Avatar src={person.avatar} />
    <Heading tag="h3" align="center">
      {person.name}
    </Heading>
    <Paragraph>
      {`${person.bio.slice(0, 300)}`}
    </Paragraph>
    <Anchor
      icon={<SocialGithubIcon />}
      href={`https://github.com/${person.github}`}
      primary
    >
      {person.github}
    </Anchor>
  </Box>
);

Contributor.propTypes = {
  person: PropTypes.shape({
    name: PropTypes.string.isRequired,
    avatar: PropTypes.string.isRequired,
    bio: PropTypes.string.isRequired,
  }),
};

export default cssModules(Contributor, styles);
