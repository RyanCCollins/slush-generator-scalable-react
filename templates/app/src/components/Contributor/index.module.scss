.contributor {
  margin-top: 0;
  margin-bottom: 1.5rem;
  background: #fff;
  border: 1px solid #dbe2e8;
  box-shadow: 0 2px 4px 0 rgba(46,61,73,0.12);
  border-radius: 0.125rem;
  transition: box-shadow 0.3s ease, border 0.3s ease;
  height: 28rem;
  margin: 10px;
  padding: 40px;
  color: #7d97ad;
  p {
    color: #7d97ad;
  }
}
