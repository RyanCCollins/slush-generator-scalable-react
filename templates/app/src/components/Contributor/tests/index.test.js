import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import Contributor from '../index';

const person = {
  name: 'Ryan Collins',
  github: 'ryanccollins',
  bio: 'Experienced Software Engineer specializing in implementing cutting-edge ' +
    'technologies in a multitude of domains, focusing on React and Front End.  ' +
    'Weekend Data Scientist and Functional Programmer.',
  avatar: 'https://avatars.githubusercontent.com/u/13810084?v=3',
};

describe('<Contributor />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Contributor person={person} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot(); // eslint-disable-line
  });
});
