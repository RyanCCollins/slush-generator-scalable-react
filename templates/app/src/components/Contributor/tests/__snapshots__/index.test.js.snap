exports[`<Contributor /> should render with default props 1`] = `
<Box
  align="center"
  announce={false}
  direction="column"
  focusable={true}
  justify="center"
  pad="none"
  responsive={true}
  size="large"
  tag="div">
  <undefined
    src="https://avatars.githubusercontent.com/u/13810084?v=3" />
  <Heading
    align="center"
    tag="h3">
    Ryan Collins
  </Heading>
  <Paragraph>
    Experienced Software Engineer specializing in implementing cutting-edge technologies in a multitude of domains, focusing on React and Front End.  Weekend Data Scientist and Functional Programmer.
  </Paragraph>
  <Anchor
    href="https://github.com/ryanccollins"
    icon={
      <SocialGithub
        a11yTitleId="social-github-title"
        responsive={true} />
    }
    primary={true}
    tag="a">
    ryanccollins
  </Anchor>
</Box>
`;
